lib/dao.js

```diff
@@ -52,6 +52,44 @@ function getIdValue(m, data) {
   return data && data[idName(m)];
 }
 
+function copyData(from, to) {
+  for (var key in from) {
+    to[key] = from[key];
+  }
+}
+
+function convertDataPropertiesByType(data) {
+  var typedData = {};
+  for (var key in data) {
+    // Convert the properties by type
+    typedData[key] = data[key];
+    if (typeof typedData[key] === 'object'
+            && typedData[key] !== null
+            && typeof typedData[key].toObject === 'function') {
+      typedData[key] = typedData[key].toObject();
+    }
+  }
+  return typedData;
+}
+
+function strictCheck(model, strict, data, inst, cb) {
```

Comment:
Please rename, function names should start with verbs. E.g. `applyStrictModeCheck`.